go-controller/pkg/cni/helper_linux.go

```diff
@@ -252,60 +252,81 @@ func setupInterface(netns ns.NetNS, containerID, ifName string, ifInfo *PodInter
 	return hostIface, contIface, nil
 }
 
+// a unique ifname is generated from the containerID by adding a variable suffix
+// for convention:
+// port representors: suffixes are _r1 (primary iface) and _r2 (secondary iface)
+// interfaces to be moved to the pod: suffixes are _1 (primary iface) and _2 (secondary iface)
+func generateIfName(containerID string, isRepresentor bool, isPrimary bool) string {
+	ifnameSuffix := "_"
+	if isRepresentor {
+		ifnameSuffix += "r"
+	}
+	if isPrimary {
+		ifnameSuffix += "1"
+	} else {
+		ifnameSuffix += "2"
+	}
+
+	// ifname max lenght is 15
+	return containerID[:(15-len(ifnameSuffix))] + ifnameSuffix
+}
+
 // Setup sriov interface in the pod
 func setupSriovInterface(netns ns.NetNS, containerID, ifName string, ifInfo *PodInterfaceInfo, deviceID string) (*current.Interface, *current.Interface, error) {
 	hostIface := &current.Interface{}
 	contIface := &current.Interface{}
-	ifnameSuffix := ""
-
 	netdevice := ifInfo.NetdevName
 
-	// 1. Move netdevice to Container namespace
-	err := moveIfToNetns(netdevice, netns)
-	if err != nil {
-		return nil, nil, err
-	}
+	primaryInterface := ifInfo.NetName == types.DefaultNetworkName
```

Comment:
please name this `isPrimaryInterface` ; it's a bit confusing to see the calls to `generateIfName(containerID, false, primaryInterface)` this way.

Reply:
will do